upup/pkg/fi/cloudup/gce/gce_cloud.go

```diff
@@ -29,24 +30,41 @@ import (
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns"
 )
 
-type GCECloud struct {
-	Compute *compute.Service
-	Storage *storage.Service
+type GCECloud interface {
```

Comment:
@andrewsykim I have updated the GCECloud struct to use the Cloud interface.  I think this will work.